### PR TITLE
Misc nvidia fixes

### PR DESCRIPTION
**Pull request by TingPing**

### CODE REVIEW

**Comment by TingPing** [comment]
@alexlarsson I don't know if I am being an idiot, but trying to run `sed` from within `nvidia-apply-extra.c` it seems I cannot run any commands?

```c
system ("/usr/bin/echo test");
```
Fails with `No such file or directory`

**Comment by alexlarsson** [comment]
@TingPing yeah, the nvidia extension is specially marked to run "without a runtime" and is completely statically linked, because its primary runtime (org.freedesktop.Platform) might not be installed when the extension is used by other runtimes.

File: nvidia-apply-extra.c [hunk 1]
```diff
@@ -265,6 +265,60 @@ find_line_offset (int fd, int skip_lines)
...
+
+  fseek (f, 0, SEEK_END);
+  fsize = ftell (f);
+  fseek (f, 0, SEEK_SET);
+
+  if ((buffer = malloc (fsize + 1)) == NULL)
+    die ("out of memory");
+
+  if ((len = fread (buffer, 1, fsize, f)) != fsize)
+    die ("failed to read file %s", path);
```
Comment (alexlarsson) [inline nvidia-apply-extra.c]: Ideally this should handle short reads and do multiple reads.

File: nvidia-apply-extra.c [hunk 2]
```diff
@@ -265,6 +265,60 @@ find_line_offset (int fd, int skip_lines)
...
+  buffer[len] = '\0';
+
+  if ((p = strstr (buffer, string)) != NULL)
+    {
```
Comment (alexlarsson) [inline nvidia-apply-extra.c]: Is there always only one instance of the string in the file?

File: nvidia-apply-extra.c (hunk 1 above)
Comment (TingPing) [reply nvidia-apply-extra.c]: This was very much a `"We only have one usage for it today and its a small file so its fine"`.

File: nvidia-apply-extra.c (hunk 2 above)
Comment (TingPing) [reply nvidia-apply-extra.c]: In this case yes.

**Pull request merged**